Gems/Terrain/Code/Source/TerrainSystem/TerrainSystem.cpp

```diff
@@ -1306,6 +1280,17 @@ void TerrainSystem::SubdivideRegionForJobs(
         subdivisionsY, maxNumJobs);
 }
 
+bool TerrainSystem::ContainedAabbTouchesXYEdge(const AZ::Aabb& outerAabb, const AZ::Aabb& innerAabb)
+{
+    return outerAabb.Contains(innerAabb) &&
+        (
+            outerAabb.GetMin().GetX() == innerAabb.GetMin().GetX() ||
+            outerAabb.GetMin().GetY() == innerAabb.GetMin().GetY() ||
+            outerAabb.GetMax().GetX() == innerAabb.GetMax().GetX() ||
+            outerAabb.GetMax().GetY() == innerAabb.GetMax().GetY()
+        );
+}
+
```

Comment:
I think this should also be checking for Z, since we're keeping that value up-to-date as well.